src/supervised/other.jl

```diff
@@ -26,12 +27,21 @@ isconvex(::PoissonLoss) = true
 isstronglyconvex(::PoissonLoss) = false
 
 # ===============================================================
-# L(target, output) = - target*ln(output) - (1-target)*ln(1-output)
+
+doc"""
+    CrossentropyLoss <: SupervisedLoss
+
+Cross-entropy loss also known as log loss and logistic loss is defined as:
+
+``L(target, output) = - target*ln(output) - (1-target)*ln(1-output)``
+"""
 
 immutable CrossentropyLoss <: SupervisedLoss end
 typealias LogitProbLoss CrossentropyLoss
 
 function value(loss::CrossentropyLoss, target::Number, output::Number)
+    target >= 0 && target <=1 || error("target must be in [0,1]")
+    output >= 0 && output <=1 || error("output must be in [0,1]")
```

Comment:
I am unsure if we should do such checks. we don't usually do them because of the nature of these functions (they get called very often in some inner loop). So normally we assumer the caller knows what he/she is doing.

I am ok with this approach as well though.